### ISSUE TITLE
GitFS broke from during upgrade from 2014.7.1 to 2014.7.5

### ISSUE THREAD

**Issue by nvx** [saltstack/salt#23116]
After upgrading my salt master from 2014.7.1 to 2014.7.5 I discovered a lot of states were no longer working with various errors such as:

```
          ID: salt-master
    Function: file.recurse
        Name: /etc/salt/master.d
      Result: False
     Comment: Recurse failed: none of the specified sources were found
     Started: 10:32:15.519512
    Duration: 89.302 ms
     Changes:
```

I discovered the root cause of this problem:

```
# salt-call cp.list_master
local:
    - ../.gitignore
    - ../_modules/mavenhelper.py
...
    - foobar/../init.sls
...
```

Notice everything seems to be prefixed with ../ - the foobar entry is actually a separate git repo with a mountpoint set as foobar, so the ../ appears, but only after the mountpoint.

Replacing /usr/lib/python2.6/site-packages/salt/fileserver/gitfs.py on the salt master with https://raw.githubusercontent.com/saltstack/salt/v2014.7.1/salt/fileserver/gitfs.py and restarting the salt master fixed the issue and everything began working as expected again.

```
# salt-call cp.list_master
local:
    - .gitignore
    - _modules/mavenhelper.py
...
    - foobar/init.sls
...
```

```
# salt --versions-report
           Salt: 2014.7.5
         Python: 2.6.6 (r266:84292, Nov 21 2013, 10:50:32)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
```

Using gitpython on RHEL6 (packages from EPEL). My gitfs remotes looks like this:

```
gitfs_provider: gitpython
gitfs_remotes:
    - git+ssh://git@example.com/salt.git
    - git+ssh://git@example.com/salt-bin.git
    - git+ssh://git@example.com/salt-formula.git
    - git+ssh://git@example.com/epel-formula.git
    - git+ssh://git@example.com/sudoers-formula.git
    - git+ssh://git@example.com/salt-abc.git:
      - mountpoint: salt://abc
    - git+ssh://git@example.com/salt-foobar.git:
      - mountpoint: salt://foobar
```

**Comment by rallytime** [comment]
@nvx Thanks for filing this report. This actually looks like a duplicate of #22987 and is likely fixed by #23120. As such, I'm going to close this one in favor of the one that was already filed. Thanks again!

**Comment by nvx** [comment]
Ah yup, sounds like the same issue indeed. Glad to here there's already a fix.
